typed/helpers.go

```diff
@@ -116,12 +97,12 @@ func resolveSchema(s *schema.Schema, tr schema.TypeRef, v *value.Value, ah atomH
 
 func deduceAtom(a schema.Atom, v *value.Value) schema.Atom {
```

Comment:
If value.Value is an interface (implying it can be nil) we probably should not pass a pointer to it, that's unnecessary and unusual.

Reply:
I think in this case, `nil` is different from not present

Reply:
Are you using `nil` to mean the same thing as the previous `Null bool`? That explains things. Why not have an explicit IsNull() function? Then you don't need this double indirection?